Refactor so that environment and engine

qihqi opened the pull request:
so that they dont depend on llama
specific stuff such as ModelArgs

FanhaiLu1 commented on jetstream_pt/engine.py:
```diff
@@ -630,7 +626,7 @@ def samples_per_slot(self) -> int:
   @property
   def max_prefill_length(self) -> int:
-    return self.param.max_seq_len
+    return self.env.max_input_sequence_length
```
Great, it's much clear now. Engineer might need to figure out and do mapping max_seq_len is max_prefill_length

FanhaiLu1 commented on jetstream_pt/engine.py:
```diff
@@ -693,24 +689,12 @@ def create_pytorch_engine(
-  if model_name == "llama":
+
+  if model_name.startswith("llama"):
+    from jetstream_pt.third_party.llama2 import model_exportable, model_args
```
Can you do import in the head of this file? Otherwise, you get below error: 
jetstream_pt/engine.py:696:4: C0415: Import outside toplevel (jetstream_pt.third_party.llama2.model_exportable, jetstream_pt.third_party.llama2.model_args) (import-outside-toplevel)

qihqi replied:
done.

FanhaiLu1 reviewed (commented):
For 'tokenize_and_pad' error, jetstream removed tokenize_and_pad in recent change, I added it back in PR https://github.com/google/JetStream/pull/70, it should pass after re-run the test.

qihqi commented:
> For 'tokenize_and_pad' error, jetstream removed tokenize_and_pad in recent change, I added it back in PR [google/JetStream#70](https://github.com/google/JetStream/pull/70), it should pass after re-run the test.

I added a command in `install_everything.sh` to pin `v0.2.0` for jetstream. We can manually update the hashes and do fixups.
